src/SFML/Window/GlContext.cpp

```diff
@@ -165,45 +165,162 @@ namespace
+        }
+
+        return sharedContext;
+    }
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Load our extensions vector with the supported extensions
+    ///
+    ////////////////////////////////////////////////////////////
+    void loadExtensions()
+    {
+        auto glGetErrorFunc    = reinterpret_cast<glGetErrorFuncType>(getFunction("glGetError"));
+        auto glGetIntegervFunc = reinterpret_cast<glGetIntegervFuncType>(getFunction("glGetIntegerv"));
+        auto glGetStringFunc   = reinterpret_cast<glGetStringFuncType>(getFunction("glGetString"));
+
+        if (!glGetErrorFunc || !glGetIntegervFunc || !glGetStringFunc)
+            return;
+
+        // Check whether a >= 3.0 context is available
+        int majorVersion = 0;
+        glGetIntegervFunc(GL_MAJOR_VERSION, &majorVersion);
+
+        auto glGetStringiFunc = reinterpret_cast<glGetStringiFuncType>(getFunction("glGetStringi"));
+
+        if (glGetErrorFunc() == GL_INVALID_ENUM || !majorVersion || !glGetStringiFunc)
+        {
+            // Try to load the < 3.0 way
+            const char* extensionString = reinterpret_cast<const char*>(glGetStringFunc(GL_EXTENSIONS));
+
+            if (extensionString)
+            {
+                extensions.clear();
+
+                do
+                {
+                    const char* extension = extensionString;
+
+                    while (*extensionString && (*extensionString != ' '))
+                        ++extensionString;
+
+                    extensions.emplace_back(extension, extensionString);
+                } while (*extensionString++);
+            }
+        }
+        else
+        {
+            // Try to load the >= 3.0 way
+            int numExtensions = 0;
+            glGetIntegervFunc(GL_NUM_EXTENSIONS, &numExtensions);
+
+            if (numExtensions)
+            {
+                extensions.clear();
+
+                for (unsigned int i = 0; i < static_cast<unsigned int>(numExtensions); ++i)
+                {
+                    const char* extensionString = reinterpret_cast<const char*>(glGetStringiFunc(GL_EXTENSIONS, i));
+
+                    if (extensionString)
+                        extensions.emplace_back(extensionString);
+                }
+            }
+        }
+    }
 
-// This per-thread variable holds the current context information for each thread
-thread_local CurrentContext currentContext;
+    // AMD drivers have issues with internal synchronization
+    // We need to make sure that no operating system context
+    // or pixel format operations are performed simultaneously
+    // This mutex is also used to protect the shared context
+    // from being locked on multiple threads
+    std::recursive_mutex mutex;
 
-// The hidden, inactive context that will be shared with all other contexts
-std::unique_ptr<ContextType> sharedContext;
+    // Supported OpenGL extensions
+    std::vector<std::string> extensions;
```

Comment:
The specification is vague on the lifetime of the returned strings. I would not put it past some driver implementations to free up the memory at an unexpected time because they assume the strings are copied by the application.